linkcheck/htmlutil/loginformsearch.py

```diff
@@ -42,16 +42,14 @@ def search_form(content, cgiuser, cgipassword):
     CGI fields. If no form is found return None.
     """
     soup = htmlsoup.make_soup(content)
-    # The value of the name attribute is case-insensitive
-    # https://www.w3.org/TR/html401/interact/forms.html#adef-name-INPUT
-    cginames = {cgiuser.lower(), cgipassword.lower()}
```

Comment:
> The keys of the form.data dictionary are case-sensitive

I'm confused. The (removed) comment above explicitely says that "the value of the name attribute is case-insensitive". Has this changed in later HTML standards? If not, why are we changing this behavior here?

I understand we might have had a KeyError inside our code, but it seems like a bad idea to remove those `lower()` call to fix it - it feels backwards?

If there's a KeyError in some cases, maybe we should have a test case to cover for that and confirm it's fixed?

In any case, color me confused by this commit (662de1818f48a49f89589f438342ca51b40f624a).

Reply:
As per the discussion linked in the comment above we were confused several times too. I think we got to the right place, minimising possible user confusion.